File: Java-SpringBoot/BE-01-InMemory/java-17-springboot-3.5/src/main/java/com/flyrank/userservice/service/UserServiceImpl.java
package com.flyrank.userservice.service;

import com.flyrank.userservice.dto.UserCreateRequest;
import com.flyrank.userservice.dto.UserResponse;
import com.flyrank.userservice.exception.UserNotFoundException;
import com.flyrank.userservice.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.UUID;

/**
 * Business logic implementation for UserService.
 */
@Service
public class UserServiceImpl implements UserService {

    private final UserRepository userRepository;

    public UserServiceImpl(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public UserResponse createUser(UserCreateRequest request) {
        UUID newId = UUID.randomUUID();
        UserResponse newRecord = new UserResponse(
                newId,
                request.name(),
                request.email(),
                request.telephone()
        );
        return userRepository.save(newRecord);
    }

    @Override
    public UserResponse getUserById(UUID id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new UserNotFoundException(id));
    }
}

File: Java-SpringBoot/BE-01-InMemory/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/userservice/service/UserServiceImpl.kt
package com.flyrank.userservice.service

import com.flyrank.userservice.dto.UserCreateRequest
import com.flyrank.userservice.dto.UserResponse
import com.flyrank.userservice.exception.UserNotFoundException
import com.flyrank.userservice.repository.UserRepository
import org.springframework.stereotype.Service
import java.util.UUID

/**
 * Business logic implementation in idiomatic Kotlin.
 */
@Service
class UserServiceImpl(
    private val userRepository: UserRepository
) : UserService {

    override fun createUser(request: UserCreateRequest): UserResponse {
        val newId = UUID.randomUUID()
        val nameDto = requireNotNull(request.name) { "name object is required" }

        val newRecord = UserResponse(
            id = newId,
            name = nameDto,
            email = request.email,
            telephone = request.telephone
        )
        return userRepository.save(newRecord)
    }

    override fun getUserById(id: UUID): UserResponse {
        return userRepository.findById(id) ?: throw UserNotFoundException(id)
    }
}
